### PR TITLE
Remove Stale label on originator issue comment

### LINKED ISSUE #9182: Issue triage automation should remove `stale` label on author feedback
See https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7807#issuecomment-1444881788

### PR DESCRIPTION
**Pull request by breedx-splk**
Resolves #9182 

Sorry @trask, I noticed that you had assigned yourself right as I pushed this. Feel free to close if you wanted to do something differently.

Also keep me honest on the spelling of `Stale`, I think it's capitalized.

### CODE REVIEW

File: .github/workflows/issue-management-feedback-label.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Issue management - remove needs feedback label
+name: Issue management - remove irrelevant labels
```
Comment (trask) [inline .github/workflows/issue-management-feedback-label.yml:1]:
```suggestion
name: Issue management - remove labels as needed
```

**Review by trask — APPROVED** [review]
> Resolves #9182
> 
> Sorry @trask, I noticed that you had assigned yourself right as I pushed this. Feel free to close if you wanted to do something differently.

no worries, appreciate the help!

> Also keep me honest on the spelling of `Stale`, I think it's capitalized.

that surprised me too, I'd be in favor of changing this to `stale` to match our other label casing, using
* https://github.com/actions/stale#stale-issue-label
* https://github.com/actions/stale#stale-pr-label

File: .github/workflows/issue-management-feedback-label.yml [hunk 2]
```diff
@@ -13,9 +13,10 @@ jobs:
...
       - uses: actions/checkout@v3
 
-      - name: Remove label
+      - name: Remove labels
         env:
           ISSUE_NUMBER: ${{ github.event.issue.number }}
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
           gh issue edit --remove-label "needs author feedback" $ISSUE_NUMBER
+          gh issue edit --remove-label "Stale" $ISSUE_NUMBER
```
Comment (breedx-splk) [inline .github/workflows/issue-management-feedback-label.yml:22]:
```suggestion
          gh issue edit --remove-label "stale" $ISSUE_NUMBER
```

**Comment by trask** [comment]
I updated the Issue Label itself also just now

**Pull request merged**